pkg/rebalancer/rebalancer.go

```diff
@@ -217,8 +217,12 @@ func (r *Rebalancer) resourcesToDelete(ctx context.Context, pods []v1.Pod) ([]re
 
 	// Sort PVCs by their creation timestamps, from the newest to the oldest to
 	// make sure that the oldest PVC is retained in case all of them are
-	// annotated.
+	// annotated. If some PVCs have the same creation timestamp, they will be
+	// sorted based on their pod name.
 	sort.Slice(resources, func(i, j int) bool {
+		if resources[i].pvc.CreationTimestamp.Equal(&resources[j].pvc.CreationTimestamp) {
+			return resources[i].pod.Name < resources[j].pod.Name
```

Comment:
if pvc creation timestamps are the same then it would return [prometheus-k8s-0, prometheus-k8s-1] and keep prometheus-k8s-1 over prometheus-k8s-0? Given how statefulsets work I would prefer to keep prometheus-k8s-0 in case of a tie. We could also look at the pod creation timestamps.
And another suggestion: it might be good to implement [`sort.Interface`](https://pkg.go.dev/sort#Interface) and verify that the sorting is as expected with unit tests.